source/validate_id.cpp

```diff
@@ -740,7 +740,7 @@ bool idUsage::isValid<SpvOpLoad>(const spv_instruction_t* inst,
            return false);
   auto pointerIndex = 3;
   auto pointer = usedefs_.FindDef(inst->words[pointerIndex]);
-  if (!pointer.first || !spvOpcodeIsPointer(pointer.second.opcode)) {
+  if (!pointer.first || !spvOpcodeIsPotentialPointer(pointer.second.opcode)) {
```

Comment:
>Well, OpBitCast doesn't say it's limited to Kernel or Addresses and it clearly states that it can bitcast pointers, so it should be possible even for Logical/shaders.

That's not what the spec currently says.  Perhaps you can file a bug and get it changed?  My impression, from talking to the spec authors, is that `OpBitCast` was deliberately forbidden from creating a pointer in the Logical addressing model.  That's why I believe the `spvOpcodeIsPointer()` should be kept, but only applied in the Logical model.

> Also, where exactly in the spec does it say that you have to use Logical for shaders?

I don't think the spec says anything like that.  It merely says what's legal if you _do_ use Logical.

> (2.16.1 is incomplete as I said).

Agreed: `OpFunctionParameter` is missing.  The spec will be updated to reflect that.  You can even include it in `spvOpcodeIsPointer()` today.

> So, it would be better to just remove spvOpcodeIsPointer/spvOpcodeIsPotentialPointer

I agree about `IsPotentialPointer`, but I believe `IsPointer` should stay, as I said above.